### ISSUE TITLE
Subfolder/directory subroutines and multi-subroutines [FEATURE REQUEST]

### ISSUE THREAD

**Issue by GhbSmwc** [JackTheSpades/SpriteToolSuperDelux#69]
**Is your feature request related to a problem? Please describe.**
I'm working on fixing my [rejected ASM/Document resource](https://github.com/GhbSmwc/SMW-Graphical-bar-display). Currently, both pixi and GPS have this nice feature that lets you have a routine be used by multiple sprites or blocks. The problem is that:

1. You cannot have multiple subroutines in the same ASM file and have sprites call a specific subroutine in that ASM file other than the first one of the ASM file, unlike uberasm tool. Uberasm tool, you can call a specific subroutine using ``JSL RoutineName_MainLabel``. Unless I'm wrong.
2. Not sure if pixi supports this, but I would like the routines to also support subdirectories just like the sprites.

**Describe the solution you'd like**
I was thinking adding an option (like at the top of the ASM file, pixi will find a string ``@@MultipleRoutinesInOneASM`` or ``DirectoryRoutines``, or something), for compatibility reasons, so that way, the routines folder doesn't look cluttered due to being required to have them all at this directory

**Additional context**
Not sure if most people would ever do this, previously I did tried using shared subroutines, but according to the moderator, Major Flare, did not like this as it requires too many steps, to insert a sprite thanks to requiring a shared subroutines patch.

**Comment by Atari2** [comment]
I am willing to add subfolder support for routines, e.g. imagine this directory structure:
```
.
└── routines/
    ├── GetDrawInfo.asm
    └── Bounce/
        └── GetDrawInfo.asm
    └── Spawn/
        └── Cluster.asm
        └── Sprite.asm
```
would generate the following routine names:
`%GetDrawInfo()`, `%BounceGetDrawInfo()`, `%SpawnCluster()` and `%SpawnSprite()` basically the folder name would get stitched to the filename (without symbols or anything).
This would also allow a bit of reorganizing of the current routines without breaking backwards compatibility.

However I'd like to keep the 1 file 1 routine rule that pixi has, it makes things a lot simpler and I don't want coders to jump randomly to a label inside of a routine, people would abuse/misuse it and I don't like it.

**Comment by Atari2** [comment]
Implemented in 7cc72d22dd039741bd6665c1ce07e6acaa81690b
